mm2src/coins_activation/src/eth_with_token_activation.rs

```diff
@@ -0,0 +1,235 @@
+use crate::{platform_coin_with_tokens::{EnablePlatformCoinWithTokensError, GetPlatformBalance,
+                                        InitTokensAsMmCoinsError, PlatformWithTokensActivationOps, RegisterTokenInfo,
+                                        TokenActivationParams, TokenActivationRequest, TokenAsMmCoinInitializer,
+                                        TokenInitializer, TokenOf},
+            prelude::*};
+use async_trait::async_trait;
+use coins::{eth::{eth_coin_from_conf_and_request_v2, Erc20Protocol, Erc20TokenActivationError,
+                  Erc20TokenActivationRequest, Erc20TokenInfo, EthActivationV2Error, EthActivationV2Request, EthCoin},
+            my_tx_history_v2::TxHistoryStorage,
+            CoinBalance, CoinProtocol, MarketCoinOps, MmCoin};
+use common::{mm_metrics::MetricsArc, Future01CompatExt};
+use futures::future::AbortHandle;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use mm2_number::BigDecimal;
+use serde::{Deserialize, Serialize};
+use serde_json::Value as Json;
+use std::collections::HashMap;
+
+impl From<EthActivationV2Error> for EnablePlatformCoinWithTokensError {
+    fn from(err: EthActivationV2Error) -> Self {
+        match err {
+            EthActivationV2Error::InvalidPayload(e) => EnablePlatformCoinWithTokensError::InvalidPayload(e),
+            EthActivationV2Error::ActivationFailed { ticker, error } => {
+                EnablePlatformCoinWithTokensError::PlatformCoinCreationError { ticker, error }
+            },
+            EthActivationV2Error::AtLeastOneNodeRequired => EnablePlatformCoinWithTokensError::AtLeastOneNodeRequired(
+                "Enable request for ETH coin must have at least 1 node".to_string(),
+            ),
+            EthActivationV2Error::CouldNotFetchBalance(e) | EthActivationV2Error::UnreachableNodes(e) => {
+                EnablePlatformCoinWithTokensError::Transport(e)
+            },
+            EthActivationV2Error::InternalError(e) => EnablePlatformCoinWithTokensError::Internal(e),
+            EthActivationV2Error::InvalidSwapContractAddr(e) => {
```

Comment:
I think mapping to `EnablePlatformCoinWithTokensError::InvalidPayload` is better idea since it's actually not valid payload. And enum values are matching as well. (`PlatformCoinCreationError` requires ticker too). Please let me know if you agree @artemii235, otherwise I will need to update it.

Reply:
> Please let me know if you agree @artemii235, otherwise I will need to update it.

@ozkanonur Yes, it's fine 🙂